openrtb/video.go

```diff
@@ -1,35 +1,40 @@
 package openrtb
 
+import "errors"
+
+// Video types annotates the parent impression as an video impression.
+// The "protocol", "sequence", "battr", "maxextended", "companionad",
+// "companiontype", and "ext" keys are unused and have been omitted.
+// See OpenRTB 2.3.1 Sec 3.2.4.
 //go:generate easyjson $GOFILE
 //easyjson:json
 type Video struct {
-	MimeTypes   []string `json:"mimes"`
-	MinDuration *int     `json:"minduration,omitempty"`
-	MaxDuration *int     `json:"maxduration,omitempty"`
-
-	// No Protocol(protocol). Use Protocols instead.
-
-	Protocols  []VideoProtocol `json:"protocols,omitempty"`
-	Width      int             `json:"w,omitempty"`
-	Height     int             `json:"h,omitempty"`
-	StartDelay int             `json:"startdelay,omitempty"`
-	Linearity  Linearity       `json:"linearity,omitempty"`
-
-	// No Sequence(sequence).
-	// No BlockedCreativeAttributes(battr).
-	// No MaxExtendedDuration(maxextended).
-
+	MIMETypes       []string         `json:"mimes"`
+	MinDuration     *int             `json:"minduration,omitempty"`
+	MaxDuration     *int             `json:"maxduration,omitempty"`
+	Protocols       []VideoProtocol  `json:"protocols,omitempty"`
+	Width           int              `json:"w,omitempty"`
+	Height          int              `json:"h,omitempty"`
+	StartDelay      int              `json:"startdelay,omitempty"`
```

Comment:
Are we okay with having this omitempty? It might be common to have StartDelay as 0 in which case it would be omitted from the Marshaled string. It goes back to what we were discussing before about using pointers for omitable fields.